service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -268,7 +268,7 @@ private JInvocation createRequest(ServiceIDL serviceIdl, JCodeModel jCodeModel,
                 if (endPoint.getRequestParamWithLimit() != null && requestParamWithLimit.equals(arg)) {
                     arg = listElementVarName;
                 }
-                if (parameter.getType().equals("String")) {
+                if (parameter.getType().equals("String") && parameter.getEncode()) {
```

Comment:
`encodeUrl` is invoked at multiple places than this instance. Is the change tested for optional queryParams as well?